scipy/stats/tests/test_kdeoth.py

```diff
@@ -589,3 +589,16 @@ def test_singular_data_covariance_gh10205():
         msg = "The data appears to lie in a lower-dimensional subspace..."
         with assert_raises(linalg.LinAlgError, match=msg):
             stats.gaussian_kde(data.T)
+
+
+def test_fewer_points_than_dimensions_gh17436():
+    # When the number of points is fewer than the number of dimensions, the
+    # the covariance matrix would be singular, and the exception tested in
+    # test_singular_data_covariance_gh10205 would occur. However, sometimes
+    # this occurs when the user passes in the transpose of what `gaussian_kde`
+    # expects. This can result in a huge covariance matrix, so bail early.
```

Comment:
Does that mean the segfault is the result of an unusually large memory allocation, or is the basis of the segfault unclear?

The general course of action here seems to make sense to me, but the discussion is sufficiently complex that I'll let a `stats` regular take a quick look first if possible.

Reply:
> Does that mean the segfault is the result of an unusually large memory allocation, or is the basis of the segfault unclear?

That's an assumption, I suppose. I guess I can look into which line is causing the segfault.